src/wallet/wallet.cpp

```diff
@@ -5129,6 +5129,11 @@ CWalletTx CWallet::CreateZerocoinSpendTransactionV3(
     selected = builder.selected;
     changes = builder.changes;
 
+    if (GetSpendAmount(tx) > Params().GetConsensus().nMaxValueSigmaSpendPerBlock) {
```

Comment:
I think we should not check like this. The transaction should already valid after created so we should change function to select mints to spend to make it never exceed limit instead.